dask/distributed/worker.py

```diff
@@ -517,6 +523,15 @@ def close(self):
     def __del__(self):
         self.close()
 
+    def heartbeat(self, pulse=5):
+        """Send a message to scheduler at a given interval"""
+        while self.status != 'closed':
+            header = {'function': 'heartbeat'}
+            payload = {}
+            future = self.pool.apply_async(self.send_to_scheduler,
+                                           args=(header, payload))
+            self._heartbeat_thread.event.wait(pulse)
```

Comment:
No, I tried to use sleep first.

When I `join` the heartbeat thread [above](https://github.com/ContinuumIO/dask/pull/375/files#diff-87cbc7893c7a5d7e693c44ad68b245aaR379), the sleep blocks until it is done. This way I can signal for it to end. Is there a better way to end the thread in the middle of a sleep call?